### DISCUSSION TITLE
Pandas pyarrow implications

### DISCUSSION THREAD

**Discussion by bamacgabhann** [anitagraser/movingpandas#309] [Ideas]
### Discussed in https://github.com/movingpandas/movingpandas/discussions/307

<div type='discussions-op-text'>

<sup>Originally posted by **bamacgabhann** May  1, 2023</sup>

4. I am still a bit unsure about the handing of datetimes in different formats, but I'm also unsure about setting them all to numpy datetime64, particularly with pandas moving towards using the pyarrow backend rather than numpy. 
5. More broadly, does anything need to be changed in mpd to allow for the pandas pyarrow backend?

<sup>Originally posted by **anitagraser** May  1, 2023</sup>
4. Unfortunately, I'm not familiar with the potential consequences of this choice.
5. That is an excellent question but I haven't investigated this yet, so I don't know.

</div>

Splitting this off for separate discussion for clarity. I had a quick look and can't see any references to dtype_backend="pyarrow" on the geopandas repo, and trying ```gdf.convert_dtypes(dtype_backend="pyarrow")``` returns ```TypeError: Cannot interpret '<geopandas.array.GeometryDtype object at 0x7f04eef1e440>' as a data type```. So I'd infer pyarrow doesn't currently support geo data, which would probably make this discussion premature. 

It would probably be best if someone who knows pandas and pyarrow better than me looks into this, but I'll have a closer look at some point soon if nobody else volunteers first.

**Comment by anitagraser** [comment]
I've performed some preliminary tests with GeoPandas and [GeoFeather](https://pypi.org/project/geofeather/) and remember running into issues with datetimes. I'll try to revisit and see if I can reproduce them. I assume GeoFeather uses the convert_dtypes logic in the background but haven't done any digging yet. GeoDataFrames without timestamps did work for me, however.
